pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -635,7 +635,19 @@ self: super: {
   spaceprobe = addBuildTool super.spaceprobe self.llvmPackages.llvm;
 
   # Tries to run GUI in tests
-  leksah = dontCheck super.leksah;
+  leksah = dontCheck (overrideCabal super.leksah (drv: {
+    executableHaskellDepends = with pkgs; [
```

Comment:
`xxxHaskellDepends` is for *Haskell* packages. Listing `gnome3.defaultIconTheme` is almost certainly wrong. I'd suggest adding those entries to the `executableSystemDepends` attribute instead. If they are specific to the test suite (i.e. they aren't required when building with `doCheck=false`), then they should go to `testSystemDepends`.
`wrapGAppsHook` is not actually a library, so this would probably have to be listed in `testToolDepends`.